src/foam/nanos/notification/NotificationSetting.js

```diff
@@ -96,7 +96,7 @@ foam.CLASS({
       name: 'authorizeOnRead',
       javaCode: `
       AuthService auth = (AuthService) x.get("auth");
-      if ( ! checkOwnership(x) && ! auth.check(x, createPermission("read")) ) throw new AuthorizationException(LACKS_READ_PERMISSION);
+      if ( ! auth.check(x, createPermission("read")) ) throw new AuthorizationException(LACKS_READ_PERMISSION);
```

Comment:
this will check for model.read.* correct?  
Should it not check for model.read.objectid?

Reply:
The createPermission("read") will resolve to "notificationsetting.read.objectid". We are granting all sme users access to notification settings.

In nanopay/src/net/nanopay/sme/groupPermissionJunctions.jrl,
p({"class":"foam.nanos.auth.GroupPermissionJunction","sourceId":"sme","targetId":"notificationsetting.read.*"})